src/main/scala/PSO/PSOFactory.scala

```diff
@@ -19,29 +21,146 @@ class PSOFactory(config: PSOConfiguration) extends Executor {
     println(s"\tOptimization Function: ${config.objectiveFunction}") 
     println(s"\tTermination Criteria: ${config.terminationCriteria}") 
 
-
     val r = new scala.util.Random(System.currentTimeMillis)
+    val bestSolution = (solutions.sortWith{
+      case (sol1, sol2) =>
+        obj(sol1) > obj(sol2)
+    }).head
 
-    val initialSolutions = (1 to particleCount).zip(Seq.fill(particleCount)(List((5.0-(-5.0))*r.nextDouble-5.0, (5.0-(-5.0))*r.nextDouble-5.0)))
-    println(initialSolutions)
+    //Create particles from initial solutions
+    val particles = solutions.to[Set].map{solution => 
+      system.actorOf(Props(classOf[Particle], 
+        Seq.fill(particleCount)(0.0), config.inertia, 
+        config.localAccel, config.neighbourhoodAccel,
+        r.nextDouble, r.nextDouble, solution, 
+        bestSolution, config.problemType, obj)
+      )
+    }
+    //connect particles based on topology
+    topology match {
+      case ParticleTopology.Ring =>
+        formRing(particles)  
+      case ParticleTopology.FullyConnected =>
+        particles.foreach(p => p ! UpdateNeighbourhood(particles.diff(Set(p))))
+      case ParticleTopology.Grid =>
+        formGrid(particles)
+      case ParticleTopology.Toroidal =>
+        formToroidal(particles)
+      case ParticleTopology.Star =>
+        particles.tail.foreach{p =>
+          p ! NewNeighbour(particles.head)
+          particles.head ! NewNeighbour(p)
+        }
+      case ParticleTopology.BinaryTree =>
+        formBinaryTree(particles.toList)
+    }
+    //termination criteria is # of iterations
+    for(i <- 1 to 1000) {
+      particles.foreach(p => p ! ComputeIteration)
+    }
+    //wait for actors to process all messages and print results
+    Thread.sleep(1000)
```

Comment:
This only runs at the very end of the execution after "termination", pretty much only there for debugging right now, so that all the Report messages are printed together at the end